### PR TITLE
UGENE-7463.  [Wd] Crash or freeze in "Extract consensus from assemblies"

### PR DESCRIPTION
**Pull request by evelinabis**
Infinite recursion
```
U2::Dashboard::setClosed()
U2::Dashboard::updateDashboard()
U2::DashboardInfoRegistry::updateDashboardInfo(const U2::DashboardInfo & newDashboardInfo)
U2::DashboardInfoRegistry::si_dashboardsChanged(const QStringList & _t1)
U2::WorkflowTabView::sl_dashboardsChanged(const QStringList & dashboardIds)
U2::Dashboard::setClosed()
...
```

### CODE REVIEW

**Review by smlnkcv — DISMISSED** [review]
No freeze, no crash

File: src/plugins/workflow_designer/src/WorkflowTabView.cpp [hunk 1]
```diff
@@ -238,8 +238,10 @@ void WorkflowTabView::sl_dashboardsChanged(const QStringList &dashboardIds) {
         }
 
         if (!dashboardInfo.opened) {
-            dashboard->setClosed();
-            removeDashboard(dashboard);
+            if (dashboardInfo.opened != dashboard->isOpened()) {
```
Comment (yalgaer) [inline src/plugins/workflow_designer/src/WorkflowTabView.cpp]: nit: I would prefer to read this check in a simpler form: `!dashboardInfo.opened && dashboard->isOpened`

**Review by yalgaer — DISMISSED** [review]
Looks good. Please merge after checking that no tests fail.

**Comment by evelinabis** [comment]
> Looks good. Please merge after checking that no tests fail.

Should I merge to master or release?

**Review by DmitriiSukhomlinov — APPROVED** [review]

**Comment by yalgaer** [comment]
Please merge into master and I will cherry-pick it into the release branch too

**Review by yalgaer — APPROVED** [review]
Please merge into the master branch and I will cherry-pick this patch into the release branch too

**Pull request merged**